### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by BenMukebo**
## In this Milestone, we use the newly acquired knowledge and refactor the previous code.

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
   - Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
   - Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @persons << Teacher.new(specialization, name, age)
+    puts 'create teacher'
+  end
```
Comment (mikethreels) [inline app.rb]: - [ ] great job cleaning up the main.rb file 👍 in this method text is printed, information is gathered and processed, this is not according to the single responsibility principle. try to split this into smaller easier to maintain methods. have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp != 'n'
+    @persons << Student.new(name, age, parent_permission)
+    puts 'create student'
+  end
```
Comment (mikethreels) [inline app.rb]: - [ ] great job cleaning up the main.rb file 👍 in this method text is printed, information is gathered and processed, this is not according to the single responsibility principle. try to split this into smaller easier to maintain methods. have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+  # '4 - Create a book',
+  def create_book
+    print('Title: ')
+    title = gets.chomp
+    print('Author: ')
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts('Book created successfully!')
+  end
```
Comment (mikethreels) [inline app.rb]: - [ ] great job cleaning up the main.rb file 👍 in this method text is printed, information is gathered and processed, this is not according to the single responsibility principle. try to split this into smaller easier to maintain methods. have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @BenMukebo and @mmsesay,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**